libs/libvtrutil/src/vtr_geometry.h

```diff
@@ -101,9 +102,17 @@ class Rect {
     //Returns true if the point is coincident with the rectangle (including the top-right edges)
     bool coincident(Point<T> point) const;
 
+    //Returns true if no points are contained in the rectangle
+    bool empty() const;
+
     friend bool operator== <>(const Rect<T>& lhs, const Rect<T>& rhs);
     friend bool operator!= <>(const Rect<T>& lhs, const Rect<T>& rhs);
 
+    template<class U>
+    friend Rect<U> operator|(const Rect<U>& lhs, const Rect<U>& rhs);
+    template<class U>
+    friend Rect<U>& operator|=(Rect<U>& lhs, const Rect<U>& rhs);
```

Comment:
Can you clarify what these operators are supposed to do? It looks like it returns the minimal enclosing rectangle of lhs and rhs. To me this isn't an immediately obvious use of the 'or' operator (one might expect it to return a RectUnion instead).

For clarity, I'd tend to shy away from defining overloaded operators unless its very obvious what the should do.

Perhaps this should just be a free function like:
```cpp
Rect<U> bounding_box(const Rect<U>& a, const Rect<U>& b);
```
which would seem to capture the intent more explicitly.

Reply:
Agreed. The operators look nice, but `bounding_box` and `expand_bounding_box` are more readable.

Reply:
Done.